hnn_core/pyramidal.py

```diff
@@ -256,8 +280,12 @@ class L2Pyr(Pyr):
     ----------
     gid : int
         The cell id.
-    p : dict
-        The parameters dictionary.
+    pos : tuple | None
+        The position of the cell.
+    geometry : dict | str | None
+        The parameters dictionary containing geometry.
+        If it's a string, then load the geometry from
+        a .geo file. If None, load the default geometry
```

Comment:
I was mulling over this. @rythorpe let me know your thoughts before I go too deep trying to implement it. Does this kind of API make sense? Or should it be rather a parameter of `set_geometry` ? Looking at these .geo files ([e.g., here](http://cns.iaf.cnrs-gif.fr/files/CELLS/pyramidal264L.geo) it seems they only define the dendrite locations and the connections. The synapses would still have to be manually defined by the user. Note I am trying to go for something that is relatively easy to implement but also serves a practical purpose ...

Reply:
This looks really good actually because of the modular flexibility it will lend to pyramidal cells. I hadn't heard of .geo files before this but I'm assuming they are the standard file type for storing dendritic geometry in nrn hoc?

I'm assuming the `geometry` argument will be passed down from higher-level `geometry` arguments of the `Network` and `NetworkBuilder` classes in order to make it accessible to the user?

Reply:
>  I hadn't heard of .geo files before this but I'm assuming they are the standard file type for storing dendritic geometry in nrn hoc?

I think so, but even if they are not ... I think the modularization would pave the way for other loaders and should be rather straigthforward. A quick google search also revealed swc files. It should also be straightforward to implement once we have this in place.

> I'm assuming the geometry argument will be passed down from higher-level geometry arguments of the Network and NetworkBuilder classes in order to make it accessible to the user?

You asked the tough question :) I don't think it would be enough to pass down a `geometry` argument from `Network` or `NetworkBuilder` because a user would still need to define mechanisms, synapses, and how other cells connect to this cell. What might work is allowing the user to define the Pyramidal cell class/objects, passing them to `Network`/`NetworkBuilder` and cloning them to create a network. Wdyt?

Reply:
>What might work is allowing the user to define the Pyramidal cell class/objects, passing them to Network/NetworkBuilder and cloning them to create a network. Wdyt?

That's an interesting idea, making the child cell classes (i.e., `L2Pyr`, `L5Pyr`, and `BasketSingle`) accessible to the user. If we do that we'll need to refactor the cell object constructor and other methods so that nrn objects related to the network are only called after the user passes them into `Network`/`NetworkBuilder`.